hack/common.sh

```diff
@@ -498,12 +498,23 @@ os::build::ldflags() {
     os::build::get_version_vars
 
     declare -a ldflags=()
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
+
+    go_version=($(go version))
+    if [[ "${go_version[2]}" < "go1.5" ]]; then
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
+      ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
+      ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
+    else
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
+      ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
+      ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
+      ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")
```

Comment:
Why we're doing this that way and not like [here](https://github.com/openshift/source-to-image/blob/master/hack/common.sh#L390-L424). Above is very error prone by copy&pasting. Please fix it as a followup.